its/src/test/java/com/sonar/it/scanner/msbuild/ScannerMSBuildTest.java

```diff
@@ -526,36 +526,28 @@ public void testXamlCompilation() throws IOException {
   }
 
   @Test
-  public void testRazorCompilation() throws IOException {
-    String localProjectKey = PROJECT_KEY + ".13";
-    ORCHESTRATOR.getServer().provisionProject(localProjectKey, "Razor");
-
-    if (TestUtils.getMsBuildPath(ORCHESTRATOR).toString().contains("14.0")) {
-      return; // This test is not supported on Visual Studio 2015
-    }
-
-    Path projectDir = TestUtils.projectDir(temp, "RazorWebApplication");
-    String token = TestUtils.getNewToken(ORCHESTRATOR);
-    ORCHESTRATOR.executeBuild(TestUtils.newScanner(ORCHESTRATOR, projectDir)
-      .addArgument("begin")
-      .setProjectKey(localProjectKey)
-      .setProjectVersion("1.0")
-      .setProperty("sonar.login", token));
-
-    TestUtils.runNuGet(ORCHESTRATOR, projectDir, "restore");
-    TestUtils.runMSBuild(ORCHESTRATOR, projectDir, "/t:Rebuild", "/nr:false");
+  public void testRazorCompilationNet2() throws IOException {
+    validateRazorProject("RazorWebApplication.net2.1");
+  }
 
-    BuildResult result = TestUtils.executeEndStepAndDumpResults(ORCHESTRATOR, projectDir, localProjectKey, token);
-    assertTrue(result.isSuccess());
+  @Test
+  public void testRazorCompilationNet3() throws IOException {
+    validateRazorProject("RazorWebApplication.net3.1");
+  }
 
-    List<Issue> issues = TestUtils.allIssues(ORCHESTRATOR);
-    List<String> ruleKeys = issues.stream().map(Issue::getRule).collect(Collectors.toList());
+  @Test
+  public void testRazorCompilationNet5() throws IOException {
+    validateRazorProject("RazorWebApplication.net5");
+  }
 
-    assertThat(ruleKeys).containsAll(Arrays.asList("csharpsquid:S1118", "csharpsquid:S1186"));
+  @Test
+  public void testRazorCompilationNet6WithoutSourceGenerators() throws IOException {
+    validateRazorProject("RazorWebApplication.net6.withoutSourceGenerators");
```

Comment:
IMO for this and the one with source generator I would also verify the content of the `csproj` file to make sure it has the expected `false` or `true`

because if by accident someone will change `<UseRazorSourceGenerator>true</UseRazorSourceGenerator>` to `false` , no test will be failing